internal/workflow.go

```diff
@@ -66,8 +66,20 @@ var (
 )
 
 type (
+	// commonChannel contains methods common to both SendChannel and ReceiveChannel
+	commonChannel interface {
+		// Name returns the name of the Channel.
+		// If the Channel was retrieved from a GetSignalChannel call, Name returns the signal name.
+		//
+		// A Channel created without an explicit name will use a generated name by the SDK and
+		// is not deterministic.
```

Comment:
```suggestion
		// A Channel created without an explicit name will use a generated name by the SDK and
		// is deterministic.
```

It is deterministic correct? (built from channel sequence)

Reply:
It is not given our normal definition for what is deterministic, and I would **not** want suers to have any expectation of the SDK generated name

Reply:
Our normal definition is "same during replay" I thought. Definitely no expectation of generated name, but it's still deterministic. Maybe we should just say "generated name by the SDK" without any additional info about determinism.